### ISSUE TITLE
FormController is not passed back to parent scope properly, when 'controller as ctrl' syntaxt is used.

### ISSUE THREAD

**Issue by snimavat** [formly-js/angular-formly#139]
When a controller used controller as syntaxt, even if name attribute is specified on formly-form, the form controller will not be set on the parent scope.

Example 
    < formly-form result="list.formData" 
    fields="list.formFields" name="list.testForm" class="form-horizontal no-margin" >
    </formly-form>

Following line in formly-form directive's compile function is the culprit

```
scope.formOnParentScope = scope[attr.name];
```

It needs to change to 
    scope.formOnParentScope = $parse(attr.name)(scope);

scope[attr.name] syntax would work for the simple cases only, not for the dot notation. eg (list.formName)

In short: i have a form name with a dot (list.formName), and i want it to be set on scope with the same property, but it wont, because it uses $scope[attr.name] in compile function, instead of $parse.

Original NgFormController handles name with a dot properly as expected.

**Comment by snimavat** [comment]
Right.. You used $eval instead of $parse.. tht too would fix the issue, great.

New version going to depend on angular 1.3.x ?
We are on 1.2.x

Thanks.

**Comment by kentcdodds** [comment]
I believe the new version should work with 1.2.x still. If it doesn't, it's a bug and should be fixed. Let me know. Thanks! Also, I highly recommend you look into upgrading to 1.3. You're missing a lot of really great stuff.

**Comment by snimavat** [comment]
Sure, i'll try when you publish a release. or from master if i get time before tht.

**Comment by kentcdodds** [comment]
I should let you know that I just pushed a breaking change to master. You may want to try pointing to 2c5aa42, or update your stuff to use `form` instead of `name` for that attribute. Good luck!
